### DISCUSSION TITLE
Separate Camera Pitch from Paint Pitch

### DISCUSSION THREAD

**Discussion by dennemark** [maplibre/maplibre-gl-js#3989] [Ideas]
Hi,

I am wondering if it would make sense to separate the camera transfrom from paint transform.
Currently the camera is limited to a pitch of 85°. If the camera is pointing higher the target point moves too close to the horizon leading to issues in loading of tiles etc.
But what if we have two transforms. 
- One that is clamped at 85° and is dealing with all the loading of tiles etc. 
- And another one that can point higher for the camera.

Actually they could be both the same transform, we just need the clamping ath the right places.

I am not sure how the shaders will support this, but I guess they should get along with a differen WVP-Matrix of the camera.

For the current use there is no reason to use a pitch of >85° for current Arc Rotate Camera. But it would allow for a camera on an animated path i.e. (already controllable by map.calculateCameraOptionsFromTo ) or possibly for a First-Person-Camera.

Kind regards

**Comment by HarelM** [comment]
Seems like a huge effort and I'm not sure I understand the value.
If the issue is with tile loading, there's already an elborate method to calculate which tiles to load, so I'm not sure splitting the camera logic and the painter logic is needed here, just to improve this method.
I can see how a pitch>85 can be used if this library is used in a game or something where you can point the camera towards the sky (first person shooter).
But it's not supported right now.
Bottom line, I guess what I'm missing here is a problem statement instead of a solution description...

**Comment by dennemark** [comment]
You are right, the main topic is, how maplibre can support a first person camera i.e. to fly over terrain or get a view from street scape, since this is currently not suppported.

I should probably better frame it. It would be good to have a separate camera pitch, not a whole separate transformer. The only thing stopping me from using a first person camera is the max pitch value. I have patched my maplibre version to support up to 180°, but this only works, if I disable all layers.

I have looked into different parts of the code already and my take is, that it would probably be enough to have a clamped pitch value at the right places.
```
/**
* limits the pitch value to 85° for tile loading
**/
get tilePitch {
  return Math.min(85, this.pitch)
}
```

I could even imagine making a PR. But would be good to know from some experienced contributors, if there are some bottlenecks with this approach. I.e. I am not sure if this approach will work well with the shaders, but I would imagine this to work.

**Comment by HarelM** [comment]
I would recommend doing a design first, understanding what needs to be change and how before opening a PR with the implementation.
Once a design is approved, a PR can follow.
I think this might be related to the following issue:
- #1016

cc: @clement-igonet
